### PR TITLE
[4.0] it shows that description of error to add new error type

### PR DESCRIPTION
**Pull request by durubayram**
Pull Request for Issue #21249  .

### Summary of Changes
by adding a new error type(COM_CONFIG_ERROR_VALIDATION) in ApplicationController.php file, you're going to be able to see description why form validation failed instead of getting an error which is 'No database selected'

### Testing Instructions

Edit the Global configuration and provide an incorrect value for "session lifetime" : 19999
(it is possible to bypass the maximum value of the input field, which is 16383)

### Expected result

The configuration is not saved and the user get a error message with the explanation of the issue which is 'Form validation failed. Please check the form.'
ekran goruntusu 83

### CODE REVIEW

**Comment by ghost** [comment]
~~I have tested this item :red_circle: unsuccessfully on 35e8d5bc21672e97e3607cb12bc44223a3991c12~~<br /><br />~~`19999` is saved and Message shows Constant.~~

Please ignore Test as for now Front- and Backend-PRs are able to test by Patchtester.

<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/21384">issues.joomla.org/tracker/joomla-cms/21384</a>.</sub>

**Comment by SharkyKZ** [comment]
Use error from model:
`$this->app->enqueueMessage($model->getError(), 'error');`

**Comment by ghost** [comment]
I have not tested this item.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/21384">issues.joomla.org/tracker/joomla-cms/21384</a>.</sub>

**Comment by ghost** [comment]
I have tested this item :red_circle: unsuccessfully on 35e8d5bc21672e97e3607cb12bc44223a3991c12<br /><br />19999 is saved and Message shows Constant.

screen shot 2018-08-03 at 15 02 46
<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/21384">issues.joomla.org/tracker/joomla-cms/21384</a>.</sub>

**Comment by Twincarb** [comment]
Is the PR missing the string being added to the language file?

**Comment by wojsmol** [comment]
@Twincarb yes

File: administrator/components/com_config/Controller/ApplicationController.php [hunk 1]
```diff
@@ -119,7 +119,8 @@ public function save()
 			 */
 
 			// Redirect back to the edit screen.
-			$this->setRedirect(Route::_('index.php?option=com_config', false));
+            $this->app->enqueueMessage(\JText::_('COM_CONFIG_ERROR_VALIDATION'), 'error');
```
Comment (brianteeman) [inline administrator/components/com_config/Controller/ApplicationController.php]: please use the namespaced code so this should be Text and not \JText

File: administrator/components/com_config/Controller/ApplicationController.php [hunk 2]
```diff
@@ -119,7 +119,8 @@ public function save()
...
+            $this->app->redirect(\JRoute::_('index.php?option=com_config', false));
```
Comment (brianteeman) [inline administrator/components/com_config/Controller/ApplicationController.php]: please use the namespaced code so this should be Route and not \JRoute

**Comment by wojsmol** [comment]
```
FILE: ...dministrator/components/com_config/Controller/ApplicationController.php
--------------------------------------------------------------------------------
FOUND 2 ERROR(S) AFFECTING 2 LINE(S)
--------------------------------------------------------------------------------
 122 | ERROR | Tabs must be used to indent lines; spaces are not allowed
 123 | ERROR | Tabs must be used to indent lines; spaces are not allowed
--------------------------------------------------------------------------------
UPGRADE TO PHP_CODESNIFFER 2.0 TO FIX ERRORS AUTOMATICALLY
--------------------------------------------------------------------------------
```

File: administrator/components/com_config/Controller/ApplicationController.php [hunk 3]
```diff
@@ -119,7 +119,8 @@ public function save()
 			 */
 
 			// Redirect back to the edit screen.
-			$this->setRedirect(Route::_('index.php?option=com_config', false));
+            $this->app->enqueueMessage(Text::_('COM_CONFIG_ERROR_VALIDATION'), 'error');
+            $this->app->redirect(\JRoute::_('index.php?option=com_config', false));
```
Comment (wojsmol) [inline administrator/components/com_config/Controller/ApplicationController.php]: use `Route` here

**Comment by wilsonge** [comment]
Codestyle fixed

**Review by brianteeman — COMMENTED** [review]
approve language changes

**Comment by brianteeman** [comment]
I have tested this item :white_check_mark: successfully on 1a3c4eb0f656275008a2bda6aa7027bdde372762<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/21384">issues.joomla.org/tracker/joomla-cms/21384</a>.</sub>

**Comment by ghost** [comment]
I have tested this item :red_circle: unsuccessfully on 1a3c4eb0f656275008a2bda6aa7027bdde372762<br /><br />Language String shown correct, but incorrect value for "session lifetime" "19999" is saved
### System information
- Nightly Build 4.0.0-alpha5-dev
- Multilingual Sample Data (French, German DE, Persian)
- Template: Cassiopeia
- macOS Sierra, 10.13.6
- Firefox 61 (64-bit)
#### CloudAccess.net
- PHP 7.1.15
- MySQLi 5.7.18-cll-lve.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/21384">issues.joomla.org/tracker/joomla-cms/21384</a>.</sub>

**Comment by obsidev** [comment]
Hi,

The "session lifetime" is not saved in the configuration ; it is coming from the session which store the form data to not loose what the user was editing.
So the user know that the configuration is not saved but its form is kept which allow him to fix what is wrong (but he need to know what is wrong)

There is a comment saying that the model should display the error message (which is not the case and we don't know why).
> The validate method enqueued all messages for us, so we just need to redirect back.

Letting the model display the errors or pushing the errors from the model is a requirement because the user must know what is going from with the provided data.
The patch fix the issue but it would require further improvements to have a good user experience.

Regards,

**Comment by rmittl** [comment]
Using "Blanks" before or after the value is accepted. 
No error message - the value after saving is 15.
Example " 18005", "18005 " is value "15" after saving

Proposed solution, trim before check.

**Comment by vijaykhollam** [comment]
I have tested this item :red_circle: unsuccessfully on 88726e023e9942930fd25b6c28fb256ffb8d175c<br /><br /># 
Language String is shown correct, but incorrect value for "session lifetime" "19999" is saved

System information
Joomla! 4.0.0-beta1-dev Development [ Amani ] 17-October-2019 20:21 GMT
Web Server : Apache 
WebServer to PHP Interface : cgi-fcgi
PHP 7.3.8
MySQL 5.7.23-cll-lve<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/21384">issues.joomla.org/tracker/joomla-cms/21384</a>.</sub>

**Comment by pravinTek** [comment]
Hi,
Now language string showing a correct message and also incorrect value for "session lifetime" "19999" is not storing in config.php 
But In the backend **UI** System -> Global Configuration still showing "19999" this value under "session lifetime".

**Comment by Quy** [comment]
This is no longer an issue.

**Pull request closed without merging**